lib/cuda/cudaRFIS012.cpp

```diff
@@ -191,8 +191,16 @@ cudaEvent_t cudaRFIS012::execute(cudaPipelineState& /*pipestate*/,
 
     rfi_S012.set_metadata(voltage.get_metadata());
     const auto& rfi_S012_meta = rfi_S012.get_metadata();
-    auto freq_upchan_factor = rfi_S012_meta->get_freq_upchan_factor();
-    auto time_downsampling_fpga = rfi_S012_meta->get_time_downsampling_fpga();
+    std::vector<int> freq_upchan_factor;
+    if (rfi_S012_meta->has_freq_upchan_factor())
+        freq_upchan_factor = rfi_S012_meta->get_freq_upchan_factor();
+    else
+        freq_upchan_factor.assign(rfi_S012_meta->get_nfreq(), 1);
+    std::vector<int> time_downsampling_fpga;
+    if (rfi_S012_meta->has_time_downsampling_fpga())
```

Comment:
Same here, this attribute should always be there.

Reply:
Good idea. Maybe it's time to define a new metadata class? I was going to clean up `chordMetadata` by removing all cruft, but I now realize that I don't know what is cruft and what is essential for some other parts of Kotekan I haven't seen.

Reply:
I would avoid having CHORMetadataN with N incremented each time we conceive of an improvement. Instead I would suggest to go the internet-rfc way and first find an implementation that works, then standardize it. Removing unused items would certainly be a good first step for this.

If this ends up being essentially a flat struct again (and no dictionary in the backend) then so be it.

Right now it would be good if the various components (stages) agreed on what to do with the current metatada.